app/src/main/java/swen_anigans/mathematicfanatic/QuizResultsActivity.java

```diff
@@ -71,14 +71,8 @@ protected void onCreate(Bundle savedInstanceState) {
 
         Star[] starValues = getStars(correct,total,starGroup.getChildCount());
         setStars(starViews,starValues);
-        if(picked == null) {
-            setRewardListeners();
-        } else {
-            LinearLayout slideUp = (LinearLayout) findViewById(R.id.slideuppanel);
-            pickedTV = ((TextView) slideUp.getChildAt(0));
-            slideUp.removeAllViews();
-            formRewardSelectedLayout(slideUp,picked);
-        }
```

Comment:
The returning from locker part was a side-effect of not having the parent-activity thing set up to force that back button functionality.

If I'm not mistaken, viewing the video directly at the quiz results, will cause the selected reward to be reset when the user stops viewing the video.  I can double check this later, but if it works without that chunk of code, I'll take your word for it.